apps/src/aichat/constants.ts

```diff
@@ -1,9 +1,21 @@
 import {EVENTS} from '@cdo/apps/lib/util/AnalyticsConstants';
+import {AiChatModelIds} from '@cdo/generated-scripts/sharedConstants';
 import modelsJson from '@cdo/static/aichat/modelDescriptions.json';
 
-import {ModelDescription, SaveType} from './types';
+import type {ModelDescription, SaveType} from './types';
 
-export const modelDescriptions: ModelDescription[] = modelsJson;
+// Build the model descriptions list from the JSON file. Only include valid model IDs.
+const modelDescriptions: ModelDescription[] = [];
+Object.values(AiChatModelIds).forEach(modelId => {
+  const description = modelsJson[modelId];
+  if (description) {
+    modelDescriptions.push({
+      id: modelId,
+      ...description,
+    });
+  }
+});
```

Comment:
I'd love to suggest doing this with a `filter` and a `map`, but I'm trying to avoid the second pass through the data that requires. There are days that I long for Rust's smarter filter that does the chaining inline so it doesn't need to completely loop through.

Anywho. Instead, I'd recommend doing it via a single reduce:

```
const modelDescriptions: ModelDescription[] = Object.values(AiChatModelIds).reduce( (bucket, modelId) => {
  const description = modelsJson[modelId];
  if (description) {
    bucket.push({
      id : modelId,
      ...description
    })
  }
  return bucket
}, []);
```

I think I typed it in right, but check my work.

That way, we're keeping the declaration and initialization more tightly coupled, and to me at least the `reduce` makes it more clearly that we're assigning data into the variable, as opposed to the `forEach` which is kinda doing the assignment via a side effect.

Also, minor note - is it possible for the `description` object to already have an `id` in it? If so, then it'd win out over the `modelId` which is assigned. I might write that as `{ ...description, id : modelId }` to be extra safe. Unless, of course, the `description`'s id is the one that should win.

Reply:
Good call! Revisiting this actually I'm inclined to turn this `modelDescriptions` constant into an ID -> description mapping as well since we do a lot of `.find`s by ID on this list once it's created. But that would require a whole bunch of other logic changes that I might save for a separate PR. In which case I actually may just revert `modelDescriptions.json` to a list to make this computation just a simple `.filter`